### PR TITLE
added eliminator.js

### PR DESCRIPTION
**Pull request by dietertaotesting**
TAO-3740 
Requires https://github.com/oat-sa/extension-tao-act/pull/462

### CODE REVIEW

File: views/js/runner/plugins/tools/answerElimination/eliminator.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
+ */
+
+// http://tao-act.lan/taoQtiTest/views/js/runner/plugins/tools/eliminator/eliminator.js?buster=3.2.0-sprint43
```
Comment (krampstudio) [inline views/js/runner/plugins/tools/answerElimination/eliminator.js]: Could you please clean this up

File: views/js/runner/plugins/tools/answerElimination/eliminator.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                    self.$choiceInteractions = $container.find('.qti-choiceInteraction');
+                    if(!self.$choiceInteractions.length) {
+                        self.hide();
+                        return;
+                    }
+                    self.show();
+                })
+                .on('unloaditem', function (){
+                    //remove all masks
+                    _.invoke(self.masks, 'destroy');
```
Comment (krampstudio) [inline views/js/runner/plugins/tools/answerElimination/eliminator.js]: This seems to be inaccurate

File: views/js/test/runner/plugins/tools/answerElimination/eliminator/test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
```
Comment (krampstudio) [inline views/js/test/runner/plugins/tools/answerElimination/eliminator/test.js]: 2016, really ;)

File: views/js/test/runner/plugins/tools/answerElimination/eliminator/test.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ */
+/**
+ * @author Dieter Raber <dieter@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'helpers',
```
Comment (krampstudio) [inline views/js/test/runner/plugins/tools/answerElimination/eliminator/test.js]: It seems that lodash and helpers are not used by the module

File: views/js/test/runner/plugins/tools/answerElimination/plugin/test.js [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
```
Comment (krampstudio) [inline views/js/test/runner/plugins/tools/answerElimination/plugin/test.js]: same

File: views/js/test/runner/plugins/tools/answerElimination/plugin/test.js [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+], function(_, helpers, runnerFactory, providerMock, eliminator) {
+    'use strict';
+
+    var providerName = 'mock';
+    runnerFactory.registerProvider(providerName, providerMock());
+
+    QUnit.module('eliminator');
+
+
+    QUnit.test('module', 3, function(assert) {
```
Comment (krampstudio) [inline views/js/test/runner/plugins/tools/answerElimination/plugin/test.js]: Please use the new `expect` syntax

```js
 QUnit.test('module', function(assert) {
      QUnit.expect(3);
```

File: views/js/test/runner/plugins/tools/answerElimination/plugin/test.js [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+                QUnit.start();
+            });
+    });
+
+
+    QUnit.asyncTest('eliminator.disable', function(assert) {
+        var runner = runnerFactory(providerName);
+        var _eliminator = eliminator(runner, runner.getAreaBroker());
+
+        _eliminator.init()
```
Comment (krampstudio) [inline views/js/test/runner/plugins/tools/answerElimination/plugin/test.js]: This is just a tip, you don't really need to nest the Promises and to check the `catch` at each level:

```js
 _eliminator
  .init()
  .then(function() {
       return  _eliminator.enable();
  })
  .then(function(){
      return  _eliminator.disable();
  })
  .catch(function(err){

  });
   
```

Comment (dietertaotesting) [inline views/js/test/runner/plugins/tools/answerElimination/plugin/test.js]: Ok, thanks for the hint!

**Comment by dietertaotesting** [comment]
The Tao main theme already contains a default style for elimination, see also newly added comment on
 views/js/runner/plugins/tools/answerElimination/eliminator.js

**Comment by hutnikau** [comment]
Please merge https://github.com/oat-sa/extension-tao-testqti/pull/722 before

**Pull request merged**